### DISCUSSION TITLE
Waveshare CAN Hat

### DISCUSSION THREAD

**Discussion by mgrouch** [bareboat-necessities/lysmarine_gen#410] [General]
Users report that these instructions for Waveshare CAN Hat work with BBN OS.

https://pysselilivet.blogspot.com/2022/05/waveshare-can-hat-with-signal-k.html?m=1

**Comment by free-x** [comment]
There are a lot of suitable Waveshare CAN HATs. 
Here we go. 
- Older RS485-CAN ( https://www.waveshare.com/wiki/RS485_CAN_HAT ) with 8MHz quartz
```
dtparam=spi=on
dtoverlay=mcp2515-can0,oscillator=8000000,interrupt=25,spimaxfrequency=1000000
```

- Older RS485-CAN ( https://www.waveshare.com/wiki/RS485_CAN_HAT ) with 12MHz quartz
```
dtparam=spi=on
dtoverlay=mcp2515-can0,oscillator=12000000,interrupt=25,spimaxfrequency=2000000
```

- RS485-CAN (B) (https://www.waveshare.com/wiki/RS485_CAN_HAT_(B))
```
dtparam=spi=on
dtoverlay=mcp2515-can0,oscillator=16000000,interrupt=25,spimaxfrequency=1000000
dtoverlay=sc16is752-spi1,int_pin=24
```
- 2Channel CAN HAT (https://www.waveshare.com/wiki/2-CH_CAN_HAT)

```
dtparam=spi=on
dtoverlay=mcp2515-can0,oscillator=16000000,interrupt=23
dtoverlay=mcp2515-can1,oscillator=16000000,interrupt=25
```

Have a lot of fun

**Reply by mgrouch** [reply]
Thanks!
